### PR TITLE
WFCORE-5897 [primary/secondary] Problematic Language usage deprecation and replacement in testsuite

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://issues.redhat.com/browse/WFCORE-5897

### CODE REVIEW

**Comment by yersan** [comment]
Hi @parsharma , I would say we need to split the changes into at least three commits as described in the main issue. We should not add any change unrelated to each point, for example, do not change elytron security user names or auth contexts, or variable names, or whatever is not forced to be changed for each main point, just focus on the three points:

1. Renaming the host-[master/slave].xml configuration files
2. Rename the default DC hostname
3. Rename properties used in the configuration files

That will ensure we are changing the tests accordingly and ensure we are not introducing wrong behaviors.

File: testsuite/domain/src/test/resources/host-configs/host-secondary.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
 <host xmlns="urn:jboss:domain:20.0"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="urn:jboss:domain:20.0 wildfly-config_20_0.xsd"
-      name="slave">
+      name="secondary">
```
Comment (yersan) [inline testsuite/domain/src/test/resources/host-configs/host-secondary.xml:26]: Check HcExtensionAndSubsystemManagementTestCase#checkSlaveReconnected, it looks like it is using a wrong host name

**Review by yersan — CHANGES_REQUESTED** [review]
In addition to the above, there are still some uses of unexpected host names in the following classes that should be part of the work done renaming the host names:

```
src/main/java/org/jboss/as/test/integration/management/extension/blocker/BlockerExtension.java:                    name = "master";
src/main/java/org/jboss/as/test/integration/management/extension/error/ErrorExtension.java:                        name = "master";
src/main/java/org/jboss/as/test/integration/management/extension/error/ErrorExtension.java:                                boolean induceOOME = false; // = context.getProcessType().isServer() && !"master".equals(targetHost.asString());
src/main/java/org/jboss/as/test/integration/management/extension/optypes/OpTypesExtension.java:                    String name = hosts.size() > 1 ? "master": hosts.iterator().next();
src/main/java/org/wildfly/test/jmx/ServiceActivatorDeploymentUtil.java:                new FilePermission(targetPath.resolve("domains").resolve("JmxControlledStateNotificationsTestCase").resolve("master").resolve("target").toAbsolutePath().toString(), "read, write"),
src/main/java/org/wildfly/test/jmx/ServiceActivatorDeploymentUtil.java:                new FilePermission(targetPath.resolve("domains").resolve("JmxControlledStateNotificationsTestCase").resolve("master").resolve("target").toAbsolutePath().toString() + File.separatorChar + '-', "read, write"),
```

**Comment by parsharma** [comment]
@yersan, I have removed ModelDescriptionConstants.MASTER and there are 3 classes using MASTER so I have changed it with String "master".

File: core-model-test/framework/src/main/java/org/jboss/as/core/model/test/CoreModelTestDelegate.java [hunk 2]
```diff
@@ -200,7 +199,7 @@ private void validateDescriptionProviders(TestModelType type, KernelServices ker
             //Big big hack to get around the fact that the tests install the host description twice
             //we're only interested in the host model anyway
             //See KnownIssuesValidator.createHostPlatformMBeanAddress
-            model = model.require(CHILDREN).require(HOST).require(MODEL_DESCRIPTION).require(MASTER);
+            model = model.require(CHILDREN).require(HOST).require(MODEL_DESCRIPTION).require("master");
```
Comment (yersan) [inline core-model-test/framework/src/main/java/org/jboss/as/core/model/test/CoreModelTestDelegate.java:202]: It looks like you missed my [comment](https://github.com/wildfly/wildfly-core/pull/5108#pullrequestreview-1044858446);  this should be replaced by "primary"

File: core-model-test/tests/src/test/java/org/jboss/as/core/model/test/host/ServerConfigTestCase.java [hunk 3]
```diff
@@ -54,7 +53,7 @@ public void testAddServerConfigBadSocketBindingGroup() throws Exception {
 
         KernelServices kernelServices = createKernelServices("host.xml");
 
-        PathAddress pa = PathAddress.pathAddress(PathElement.pathElement(HOST, MASTER), PathElement.pathElement(SERVER_CONFIG, "server-four"));
+        PathAddress pa = PathAddress.pathAddress(PathElement.pathElement(HOST, "master"), PathElement.pathElement(SERVER_CONFIG, "server-four"));
```
Comment (yersan) [inline core-model-test/tests/src/test/java/org/jboss/as/core/model/test/host/ServerConfigTestCase.java:56]: It looks like you missed my [comment](https://github.com/wildfly/wildfly-core/pull/5108#pullrequestreview-1044858446);  this and the others on the same class should be replaced by "primary"

File: host-controller/src/main/java/org/jboss/as/host/controller/discovery/S3Discovery.java [hunk 4]
```diff
@@ -130,7 +129,7 @@ public int compare(DomainControllerData data, DomainControllerData otherData) {
                     }
                     return -1;
                 }});
-            writeToFile(data, MASTER);
+            writeToFile(data, "master");
```
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/discovery/S3Discovery.java:132]: This one is OK, we will handle S3Discovery.java in a different Jira.

**Review by yersan — CHANGES_REQUESTED** [review]
Hi @parsharma , would you mind reviewing [my last comment](https://github.com/wildfly/wildfly-core/pull/5108#pullrequestreview-1044858446)? it seems you missed it. 

Let me know when it's ready to review; in addition, check if your custom integration Jobs are passing after applying the changes, thanks

**Comment by parsharma** [comment]
> CoreModelTestDelegate
> ServerConfigTestCase

These two classes are not the part of test suite so I think we should not change it to primary.

**Comment by parsharma** [comment]
@yersan PR is ready for review. Only two test cases are failing in the integration jobs [CI Job](https://ci.wildfly.org/buildConfiguration/WildFlyCore_PullRequest_WildFlyCoreFullIntegrationLinuxJdk11_2/316451?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true)

**Comment by yersan** [comment]
@parsharma The integration job failures need to be corrected on the WildFly full PR.

**Comment by parsharma** [comment]
Wildfly PR: https://github.com/wildfly/wildfly/pull/15613

**Review by yersan — CHANGES_REQUESTED** [review]
> @yersan
> 
> > CoreModelTestDelegate
> > ServerConfigTestCase
> 
> These two classes are not the part of test suite so I think we should not change it to primary.

These tests are part of `wildfly-core-model-test-framework`, which is used in the testsuite (both in WildFly Full and Core) we have to change them together with this PR to avoid the need of doing another integration between changes in WildFly Core that requires changes in full.

**Review by yersan — APPROVED** [review]
LGTM, I'm just only waiting to see the result of an integration Job in Windows. 
@parsharma would you mind adding the Jira number to all the commits? Having a Jira number there is very useful when you are looking for the reason for a change in the GitHub history. 

There is no need to trigger the CI Integration jobs again.

Linux: https://ci.wildfly.org/viewLog.html?buildId=324460
Windows: https://ci.wildfly.org/viewLog.html?buildId=324659

**Comment by parsharma** [comment]
@yersan changes are done. Thanks

**Pull request merged**